src/attestedclaim/AttestedClaim.ts

```diff
@@ -8,12 +8,71 @@
  * @preferred
  */
 
-import Attestation from '../attestation/Attestation'
-import RequestForAttestation from '../requestforattestation/RequestForAttestation'
-import IAttestedClaim from '../types/AttestedClaim'
+import Attestation, {
+  compressAttestation,
+  decompressAttestation,
+} from '../attestation/Attestation'
+import RequestForAttestation, {
+  compressRequestForAttestation,
+  decompressRequestForAttestation,
+} from '../requestforattestation/RequestForAttestation'
+import IAttestedClaim, { CompressedAttestedClaim } from '../types/AttestedClaim'
 import IAttestation from '../types/Attestation'
 import IRequestForAttestation from '../types/RequestForAttestation'
 
+function attestedClaimErrorCheck(attestedClaim: IAttestedClaim): void {
```

Comment:
I took the check from inside the `AttestedClaim` constructor and moved it as a separate function.

As I used it to check inside the `compressAttestedClaim` and the constructor.